src/metabase/query_processor/middleware/update_used_cards.clj

```diff
@@ -35,6 +47,8 @@
   (mu/fn [query :- ::qp.schema/query
           rff   :- ::qp.schema/rff]
     (letfn [(rff* [metadata]
-             (update-used-cards!* (set (lib.metadata/invoked-ids (qp.store/metadata-provider) :metadata/card)))
-             (rff metadata))]
+              (doseq [card-id (set (lib.metadata/invoked-ids (qp.store/metadata-provider) :metadata/card))]
```

Comment:
nit: I'd rather use `distinct` here. It doesn't make much difference, but it's bit clearer and more idiomatic - sets are not designed for efficient iteration.

Reply:
you got me, updated to use `distinct`